lib/Sema/TypeCheckGeneric.cpp

```diff
@@ -170,7 +170,7 @@ Type CompleteGenericTypeResolver::resolveDependentMemberType(
   // Resolve the base to a potential archetype.
   auto basePA = Builder.resolveArchetype(baseTy);
   assert(basePA && "Missing potential archetype for base");
-  basePA = basePA->getRepresentative();
+  basePA = basePA->getArchetypeAnchor();
```

Comment:
Doug had suggested it while I was trying to work through some same type issues. It doesn't appear to be necessary, so I can remove it.